src/Support/SimplePaginationType.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rebing\GraphQL\Support;
+
+use GraphQL\Type\Definition\ObjectType;
+use GraphQL\Type\Definition\Type as GraphQLType;
+use Illuminate\Pagination\Paginator;
+use Illuminate\Support\Collection;
+use Rebing\GraphQL\Support\Facades\GraphQL;
+
+class SimplePaginationType extends ObjectType
+{
+    public function __construct(string $typeName, string $customName = null)
+    {
+        $name = $customName ?: $typeName.'SimplePagination';
+
+        $config = [
+            'name'   => $name,
+            'fields' => $this->getPaginationFields($typeName),
+        ];
+
+        $underlyingType = GraphQL::type($typeName);
+        if (isset($underlyingType->config['model'])) {
+            $config['model'] = $underlyingType->config['model'];
+        }
+
+        parent::__construct($config);
+    }
+
+    /**
+     * @param string $typeName
+     *
+     * @return array<string, array<string,mixed>>
+     */
+    protected function getPaginationFields(string $typeName): array
+    {
+        return [
+            'data'           => [
+                'type'        => GraphQLType::listOf(GraphQL::type($typeName)),
```

Comment:
I don't think this field can ever return `null`:
```suggestion
                'type'        => Type::nonNull(GraphQLType::listOf(GraphQL::type($typeName))),
```

Reply:
Type::nonNull seems not work with ListOf. ListOf can return an empty list. I added a new test case for this. Or you mean this?
`GraphQLType::listOf(Type::nonNull(GraphQL::type($typeName)))`

Reply:
Good question!

> Type::nonNull seems not work with ListOf

Oh, but it should!

In this case I meant that the value returned by `resolve` cannot be `null`, ever, because even the closure has the typehint `Collection`.

This means there will always be a collection returned, never null, so wrapping the existing `GraphQLType::listOf(GraphQL::type($typeName)),` into non-null like `GraphQLType::nonNull(GraphQLType::listOf(GraphQL::type($typeName))),` should be possible!

(sorry I see in my feedback I just write `Type::nonNull` which is usually the alias used, but in this class it's `GraphQLType`, so maybe that's why it may look confusing).

> ListOf can return an empty list

Exactly! But "empty list" !== "null"!

> Or you mean this? `GraphQLType::listOf(Type::nonNull(GraphQL::type($typeName)))`

I would love to (always good to avoid null values, if possible), but I think a collection _containing_ null values is technically allowed (however unlikely), so therefore sadly no, didn't meant this one.

Reply:
Hi @mfn 
Thanks for taking your time to review my code.

I updated the code into this:
`GraphQLType::nonNull(GraphQLType::listOf(GraphQL::type($typeName)))`
The tests failed.
I checked in the file tests/Database/SelectFields/PrimaryKeyTests/PrimaryKeySimplePaginationQuery.php
`
/** @var SelectFields $selectFields */
        $selectFields = $getSelectFields();
$selectFields->getSelect()
`
only return `posts.id`. If I remove GraphQLType::nonNull, it will return` posts.id, posts.title`. I think there is a bug in src/Support/SelectFields.php or miss check somewhere. Can you help me? I'm not familiar with this. Thanks very much.

Reply:
TBH I've no idea.

I don't consider it a showstopper and I also didn't write the existing `PaginationType` which works _exactly_ the same way, so I actually do feel bad for side-tracking this 😞